packages/terra-doc-template/src/DocTemplate.jsx

```diff
@@ -0,0 +1,106 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import PropsTable from 'terra-props-table';
+import Markdown from 'terra-markdown';
+import IndexExampleTemplate from './IndexPageTemplate';
+
+const propTypes = {
+  /**
+   * The given component's npm package name
+   */
+  packageName: PropTypes.string,
+  /**
+   * The given component's readme file imported to a string
+   */
+  readme: PropTypes.string,
+  /**
+   * All of the example(s) that will be displayed. An empty array is supported.
+   * ```
+   * title: The title of the example
+   * description: A description of the example to be displayed below the title
+   * example: The example to be displayed
+   * source: The source code of the example
+   * ```
+   */
+  examples: PropTypes.arrayOf(
+    PropTypes.shape(
+      {
+        title: PropTypes.string,
+        description: PropTypes.string,
+        example: PropTypes.element,
+        source: PropTypes.string,
+      },
+    ),
+  ),
+  /**
+   * All of the props table(s) that will be displayed. An empty array is supported.
+   * ```
+   * componentSrc: The source code of the component
+   * componentName: The name of the component
+   * ```
+   */
+  propsTables: PropTypes.arrayOf(
+    PropTypes.shape(
+      {
+        componentSrc: PropTypes.string,
+        componentName: PropTypes.string,
+      },
+    ),
+  ),
+};
+
+const defaultProps = {
+  packageName: '',
+  readme: '',
+  examples: [],
+  propsTables: [],
+};
+
+const DocTemplate = (props) => {
+  const { packageName, readme, examples, propsTables } = props;
+  let id = 0;
+
+  // Assign unique identifiers to use as keys later
+  for (let i = 0; i < examples.length; i += 1) {
+    examples[i].id = id;
+    id += 1;
+  }
+
+  for (let i = 0; i < propsTables.length; i += 1) {
+    propsTables[i].id = id;
+    id += 1;
+  }
+
+  const version = `[![NPM version](http://img.shields.io/npm/v/${packageName}.svg)](https://www.npmjs.org/package/${packageName})`;
+
+  return (
+    <div>
+      {packageName && <Markdown src={version} />}
+      {readme && <Markdown src={readme} />}
+
+      {examples.length > 0 && <h1 style={{ paddingBottom: '0.3em', borderBottom: '1px solid #eaecef' }}>Examples</h1>}
```

Comment:
Minor - do we want to split out the styles here? To keep styling consistency with other packages, we can move these to a separate `DocTemplate.scss` file, then localize the styles with css modules (similar to [here](https://github.com/cerner/terra-core/blob/master/packages/terra-status/src/Status.scss)).